onmt/translate/translator.py

```diff
@@ -1228,3 +1229,94 @@ def _score_target(
         gold_scores = gold_scores.sum(dim=0).view(-1)
 
         return gold_scores
+
+
+class ScoringPreparator(nn.Module):
+    import torch.nn as nn
+    from sacrebleu import corpus_bleu
+
+    def __init__(self, fields, opt):
+        self.fields = fields
+        self.opt = opt
+        import pyonmttok
+        self.tgt_tokenizer = pyonmttok.Tokenizer(
+            **self.opt.tgt_onmttok_kwargs)
+
+    def tokenize_batch(self, batch_side, side):
+        field = dict(self.fields)[side].base_field
```

Comment:
Aren't the fields already a `dict` here?

Reply:
This is the case indeed.
This line could also be updated :
https://github.com/OpenNMT/OpenNMT-py/blob/fc3409ce08785915728dac9e9c9fb49f73027917/onmt/trainer.py#L34